sycl/plugins/level_zero/pi_level_zero.cpp

```diff
@@ -3386,6 +3424,13 @@ static void recycleEventCommandList(pi_event Event) {
       }
     }
   }
+
+  // Release the kernel associated with this event if there is one.
+  if (Event->CommandType == PI_COMMAND_TYPE_NDRANGE_KERNEL &&
+      Event->CommandData) {
+    piKernelRelease((pi_kernel)(Event->CommandData));
+    Event->CommandData = nullptr;
```

Comment:
```suggestion
    Event->CommandData = nullptr;
    piKernelRelease(pi_cast<pi_kernel>(Event->CommandData));
```

I am not sure if data race is possibly here today, but setting this to null before actually using it seems safer.

Reply:
The change you suggest is incorrect.  If done the way you suggest, that would just pass nullptr to piKernelRelease always.  There cannot be a race.  The event member Event->CommandData is protected from race by the lock of Queue->PiQueuMutex, the same as Event->ZeCommandList is in the lines just above this.

Reply:
Of course, I meant to save, then nullify, and then operate on the saved one. I see the mutex and it does look sufficent.